source/material_model/visco_plastic.cc

```diff
@@ -191,15 +197,34 @@ namespace aspect
           // Convert friction angle from degrees to radians
           double phi = angles_internal_friction[j] * numbers::PI/180.0;
 
+          // Assing cohesions to a new variable
+          double coh = cohesions[j];
+
+          // Strain weakening
+          double strain_ii = 0;
+          if (use_strain_weakening == true)
+
+            // Calculate second strain invariant
+            strain_ii = ( (dim==3)
```

Comment:
Is this different from the definition of the `second_invariant` function that is for example used in diffusion_dislocation.cc:128? The definition of that function is in deal.II/include/deal.II/base/symmetric_tensor.h:2269 and 2285. If yes maybe explain it here. If not: best use those other functions.